image-prune

```diff
@@ -199,6 +202,30 @@ class LocalImageDirectory(ImageCache):
         return free >= self.space_threshold
 
 
+class S3ImageStore(ImageCache):
+    def __init__(self, url, max_gb=80):
+        self.url = url
+        self.max_bytes = (max_gb * 1000 * 1000 * 1000)
+
+    def list_files(self):
+        result = s3.list_bucket(self.url)
+
+        # make sure this gets done before we start iterating
+        self.sizes = dict(s3.parse_list(result, "Key", "Size"))
+
+        RFC3339ish = '%Y-%m-%dT%H:%M:%S.%f%z'
+        for name, stamp in s3.parse_list(result, "Key", "LastModified"):
+            yield name, datetime.datetime.strptime(stamp, RFC3339ish).timestamp()
```

Comment:
Does it has to be that specific, or could it just use datetime.isoformat()?

Reply:
`datetime.isoformat()` is only for printing.  I did quite some searching and there's no good python standard library function for ISO8601/3339 *parsing*.

Reply:
Argh, "p", not "f", sorry!